plugins/hls-refactor-plugin/src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -1482,17 +1485,62 @@ suggestNewImport df packageExportsMap ps fileContents Diagnostic{_message}
         >>= (findImportDeclByModuleName hsmodImports . T.unpack)
         >>= ideclAs . unLoc
         <&> T.pack . moduleNameString . unLoc
+  , -- tentative workaround for detecting qualification in GHC 9.4
+    -- FIXME: We can delete this after dropping the support for GHC 9.4
+    qualGHC94 <-
+        guard (ghcVersion == GHC94)
```

Comment:
alternatively we could use some CPP. that would let us only have the dependencies conditionally

Reply:
Why CPP when this works just as well? CPP is always akward, to format, to read... I think this is great the way it is.

Reply:
The only reason would be to make the extra dependencies conditional, but I agree it's probably not worth it.

Reply:
As I decided to keep dependency for `regex-applicative` as-is, I also left the import unchanged.
I don't like conditional import so much as it also makes `add import` code action behaving awkwardly; e.g. adding import inside CPP conditional block in some cases even when meant as an unconditional import!

Reply:
Oops, this thread was meant for logic not import - sorry for my misreading. But again, I don't like CPP so much when we can use source-level conditional instead with a little overhead. In this case I expect (but not verified) compilers to optimise away this conditional.